app/src/main/java/org/mozilla/fenix/home/recentvisits/view/RecentVisitsHeaderViewHolder.kt

```diff
@@ -17,18 +39,83 @@ import org.mozilla.fenix.utils.view.ViewHolder
  * interactions.
  */
 class RecentVisitsHeaderViewHolder(
-    view: View,
+    composeView: ComposeView,
+    viewLifecycleOwner: LifecycleOwner,
     private val interactor: RecentVisitsInteractor
-) : ViewHolder(view) {
+) : ComposeViewHolder(composeView, viewLifecycleOwner) {
 
     init {
-        val binding = RecentVisitsHeaderBinding.bind(view)
-        binding.showAllButton.setOnClickListener {
-            interactor.onHistoryShowAllClicked()
+        val horizontalPadding =
+            composeView.resources.getDimensionPixelSize(R.dimen.home_item_horizontal_margin)
+        composeView.setPadding(horizontalPadding, 0, horizontalPadding, 0)
+    }
+
+    @Composable
+    override fun Content() {
+        Column {
+            Spacer(modifier = Modifier.height(40.dp))
+
+            RecentVisitsHeader(stringResource(R.string.history_metadata_header_2)) {
+                interactor.onHistoryShowAllClicked()
+            }
+
+            Spacer(Modifier.height(16.dp))
         }
     }
 
     companion object {
-        const val LAYOUT_ID = R.layout.recent_visits_header
+        val LAYOUT_ID = View.generateViewId()
+    }
+}
+
+/**
+ * Recent visits homepage header.
+ *
+ * @param text The header string.
+ * @param onShowAllClick Action to take when show all is clicked.
```

Comment:
nit: the kdoc for `onShowAllClick` may want to mention the lambda/action is optional